crates/hir_ty/src/infer/path.rs

```diff
@@ -230,6 +230,7 @@ impl<'a> InferenceContext<'a> {
             self.trait_env.clone(),
             krate,
             &traits_in_scope,
+            None,
```

Comment:
There is some inconsistency right now because `LookupMode::Path` mode doesn't pass `visible_from_module` down to `is_valid_candidate`. The idea was to preserve the old resolver resolution behavior, however I am not sure...

Reply:
Okay the latest commit passes `visible_from_module` down to `iterate_inherent_methods`.